tests/operator_tests.py

```diff
@@ -476,13 +467,79 @@ def test_pda_power(self):
         ex = np.power(n, [2, 3, 4])
         self.assertListEqual(p.to_list(), ex.tolist())
 
+        # Test a singleton with and without a Boolean argument
+        n = np.array([7])
+        a = ak.array(n)
+
+        p_True = ak.power(a, 3, True)
+        p_Imply = ak.power(a, 3)
+        p_False = ak.power(a, 3, False)
+        self.assertListEqual(p_True.to_list(), p_Imply.to_list())
+        self.assertListEqual(p_False.to_list(), a.to_list())
+
+        # Test an with and without a Boolean argument, all the same
+        n = np.array([0, 0.0, 1, 7.0, 10])
+        a = ak.array(n)
+
+        truth_True = ak.array([True, True, True, True, True])
+        truth_False = ak.array([False, False, False, False, False])
+        p_True = ak.power(a, 3, truth_True)
+        p_Imply = ak.power(a, 3)
+        p_False = ak.power(a, 3, truth_False)
+        self.assertListEqual(p_True.to_list(), p_Imply.to_list())
+        self.assertListEqual(p_False.to_list(), a.to_list())
+
+        # Test a singleton with and without a Boolean argument, mixed
+        n = np.arange(10)
+        a = ak.array(n)
+        n_Calculated = np.array([0, 1, 4, 3, 16, 5, 36, 7, 64, 9])
+        a_Calculated = ak.array(n_Calculated)
+
+        truth = ak.array([True, False, True, False, True, False, True, False, True, False])
+        p = ak.power(a, 2, truth)
+        self.assertListEqual(p.to_list(), a_Calculated.to_list())
```

Comment:
instead of hardcoding the arrays, i think we could test the same functionality by doing
```python
a = ak.arange(10)
self.assertListEqual([i if i%2 else i**2 for i in range(10)], ak.power(a, 2, a%2==0).to_list())
```